drop-boxes/register-omero-metadata/backendinterface.py

```diff
@@ -0,0 +1,381 @@
+
+           Args:
+               conn: Established Connection to the OMERO Server via a BlitzGateway
+               openbis_project_id: Id specifying the project information stored on OpenBIS
+               openbis_sample_id: Id specifying the sample information stored on OpenBIS
+           Returns:
+               omero_dataset_id:  Id specifying the dataset information stored on OMERO
+
+           """
+    omero_dataset_id = -1
+    found_id = False
+
+    my_exp_id = conn.getUser().getId()
+    default_group_id = conn.getEventContext().groupId
+
+    for project in conn.getObjects("Project"):
+
+        if found_id:
+            break
+
+        if project.getName() == openbis_project_id:
+            for dataset in project.listChildren():
+
+                if dataset.getName() == openbis_sample_id:
+                    omero_dataset_id = dataset.getId()
+
+                    found_id = True
+                    break
+
+    return omero_dataset_id
+
+def register_image_file(file_path, project_id, sample_id, usr, pwd, host, port=4064):
+    """
+    This function imports an image file to an omero server using the OMERO.cli (using Bio-formats)
+    This function assumes the OMERO.cli is installed
+    Example:
+        register_image_file("data/test_img.nd2", "project_x", "sample_y",
+         "joe_usr", "joe_pwd", "192.168.2.2")
+    Args:
+        file_path (string): the path to the fastq file to validate
+        project_id (string): the corresponding project ID in openBIS server
+        sample_id (string): the corresponding sample ID in openBIS server
+        usr (string): username for the OMERO server
+        pwd (string): password for the OMERO server
+        host (string): OMERO server address
+        port (int): OMERO server port
+    Returns:
+        list of strings: list of newly generated omero IDs for registered images
+                (a file can contain many images)
+    """
+
+    import subprocess
+
+    image_ids = []
+
+    conn = omero_connect(usr, pwd, host, str(port))
+    ds_id = get_omero_dataset_id(conn, project_id, sample_id)
+
+    if ds_id != -1:
+
+        cmd = "omero-importer -s " + host + " -p " + str(port) + " -u " + usr + " -w " + pwd + " -d " + str(int(ds_id)) + " " + file_path
+        print("----cmd: " + cmd)
+
+        proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True,
+                            universal_newlines=True)
+
+        std_out, std_err = proc.communicate()
+
+        print("std_out: " + std_out)
+        print("std_err: " + std_err)
+        print("return_code: " + str(proc.returncode))
+
+        if int(proc.returncode) == 0:
+
+            print("-->" + std_out)
+
+            fist_line = std_out.splitlines()[0]
+            image_ids = fist_line[6:].split(',')
```

Comment:
this looks dangerous

Reply:
indeed. but this was the code that has been used and we reference this for the publication. 

Regex would be a safer way to go here. Would you mind and open an issue @wow-such-code such that we can keep track of it?